### PR TITLE
xlat: move '%{strlen: ...} to async

### PR DESCRIPTION
**Pull request by jpereira**

### CODE REVIEW

File: src/lib/server/xlat_func.c [hunk 1]
```diff
@@ -199,12 +199,25 @@ int xlat_fmt_to_cursor(TALLOC_CTX *ctx, fr_cursor_t **out,
...
+	fr_value_box_t	*vb;
+
+	MEM(vb = fr_value_box_alloc(ctx, FR_TYPE_UINT32, NULL, false));
+
+	vb->vb_int32 = 0;
+
+	if (*in) {
+		char const *str = (*in)->vb_strvalue;
+
+		vb->vb_uint32 = strlen(str);
```
Comment (arr2036) [inline src/lib/server/xlat_func.c]: strlen returns a size_t, so the value box should really be FR_TYPE_SIZE

File: src/lib/server/xlat_func.c [hunk 2]
```diff
@@ -199,12 +199,25 @@ int xlat_fmt_to_cursor(TALLOC_CTX *ctx, fr_cursor_t **out,
...
-			   UNUSED REQUEST *request, char const *fmt)
+static xlat_action_t xlat_strlen(TALLOC_CTX *ctx, fr_cursor_t *out,
+				 UNUSED REQUEST *request, UNUSED void const *xlat_inst, UNUSED void *xlat_thread_inst,
+				 fr_value_box_t **in)
 {
-	snprintf(*out, outlen, "%u", (unsigned int) strlen(fmt));
-	return strlen(*out);
+	fr_value_box_t	*vb;
+
+	MEM(vb = fr_value_box_alloc(ctx, FR_TYPE_UINT32, NULL, false));
```
Comment (arr2036) [inline src/lib/server/xlat_func.c]: You also need to concat the input strings together

**Comment by jpereira** [comment]
@arr2036 done.

File: src/lib/server/xlat_func.c [hunk 3]
```diff
@@ -199,12 +199,34 @@ int xlat_fmt_to_cursor(TALLOC_CTX *ctx, fr_cursor_t **out,
 /** Print length of its RHS.
  *
  */
-static ssize_t xlat_strlen(UNUSED TALLOC_CTX *ctx, char **out, size_t outlen,
-			   UNUSED void const *mod_inst, UNUSED void const *xlat_inst,
-			   UNUSED REQUEST *request, char const *fmt)
+static xlat_action_t xlat_strlen(TALLOC_CTX *ctx, fr_cursor_t *out,
+				 UNUSED REQUEST *request, UNUSED void const *xlat_inst,
```
Comment (arr2036) [inline src/lib/server/xlat_func.c]: Request should be marked as UNUSED if you're using debugging macros, because they get logging info from the request

Comment (arr2036) [reply src/lib/server/xlat_func.c]: I'll merge this one and fix it on my side

**Pull request merged**